### ISSUE TITLE
GTF file causing failure in STAR 2.4.1d

### ISSUE THREAD

**Issue by diekhans** [alexdobin/STAR#76]
I am getting:
  terminate called after throwing an instance of 'std::out_of_range'
  what():  vector::_M_range_check

when building an index.  The files to reproduce the faire are here:
   http://hgwdev.cse.ucsc.edu/~markd/bugreports/start-gtf-bugreport.tar.gz

The showBug script has the command line to cause the failure. 

the command line is:
    ../STAR  --genomeFastaFiles data/chr22.fa --runMode genomeGenerate --runThreadN 4 --genomeDir index --outTmpDir tmp --sjdbGTFfile data/debug.gtf --sjdbOverhang 74

This is chr22 from GRCh38 using the UCSC naming convention.  The GTF is from GENCODE V23.

This aborts in loadGTF.cpp on ```trOut << transcriptID.at(trid)```

transcriptID is empty and trid is a very huge number (looks uninitialized)
```
chr22	ENSEMBL	gene	10736171	10736283	.	-	.	gene_id "ENSG00000277248.1"; gene_type "snRNA"; gene_status "NOVEL"; gene_name "U2"; level 3;
chr22	ENSEMBL	transcript	10736171	10736283	.	-	.	gene_id "ENSG00000277248.1"; transcript_id "ENST00000615943.1"; gene_type "snRNA"; gene_status "NOVEL"; gene_name "U2"; transcript_type "snRNA"; transcript_status "NOVEL"; transcript_name "U2.14-201"; level 3; tag "basic"; transcript_support_level "NA";
chr22	ENSEMBL	exon	10736171	10736283	.	-	.	gene_id "ENSG00000277248.1"; transcript_id "ENST00000615943.1"; gene_type "snRNA"; gene_status "NOVEL"; gene_name "U2"; transcript_type "snRNA"; transcript_status "NOVEL"; transcript_name "U2.14-201"; exon_number 1; exon_id "ENSE00003736336.1"; level 3; tag "basic"; transcript_support_level "NA";
chr22	ENSEMBL	gene	10936023	10936161	.	-	.	gene_id "ENSG00000274237.1"; gene_type "miRNA"; gene_status "NOVEL"; gene_name "CU459211.1"; level 3;
chr22	ENSEMBL	transcript	10936023	10936161	.	-	.	gene_id "ENSG00000274237.1"; transcript_id "ENST00000618365.1"; gene_type "miRNA"; gene_status "NOVEL"; gene_name "CU459211.1"; transcript_type "miRNA"; transcript_status "NOVEL"; transcript_name "CU459211.1-201"; level 3; tag "basic"; transcript_support_level "NA";
chr22	ENSEMBL	exon	10936023	10936161	.	-	.	gene_id "ENSG00000274237.1"; transcript_id "ENST00000618365.1"; gene_type "miRNA"; gene_status "NOVEL"; gene_name "CU459211.1"; transcript_type "miRNA"; transcript_status "NOVEL"; transcript_name "CU459211.1-201"; exon_number 1; exon_id "ENSE00003712615.1"; level 3; tag "basic"; transcript_support_level "NA";
chr22	ENSEMBL	gene	11065974	11067346	.	-	.	gene_id "ENSG00000280363.1"; gene_type "protein_coding"; gene_status "KNOWN"; gene_name "CU104787.1"; level 3;
chr22	ENSEMBL	transcript	11065974	11067346	.	-	.	gene_id "ENSG00000280363.1"; transcript_id "ENST00000623473.1"; gene_type "protein_coding"; gene_status "KNOWN"; gene_name "CU104787.1"; transcript_type "protein_coding"; transcript_status "KNOWN"; transcript_name "CU104787.1-201"; level 3; protein_id "ENSP00000485388.1"; tag "basic"; transcript_support_level "5"; tag "appris_principal_1";
chr22	ENSEMBL	exon	11067335	11067346	.	-	.	gene_id "ENSG00000280363.1"; transcript_id "ENST00000623473.1"; gene_type "protein_coding"; gene_status "KNOWN"; gene_name "CU104787.1"; transcript_type "protein_coding"; transcript_status "KNOWN"; transcript_name "CU104787.1-201"; exon_number 1; exon_id "ENSE00003759492.1"; level 3; protein_id "ENSP00000485388.1"; tag "basic"; transcript_support_level "5"; tag "appris_principal_1";
chr22	ENSEMBL	CDS	11067335	11067346	.	-	0	gene_id "ENSG00000280363.1"; transcript_id "ENST00000623473.1"; gene_type "protein_coding"; gene_status "KNOWN"; gene_name "CU104787.1"; transcript_type "protein_coding"; transcript_status "KNOWN"; transcript_name "CU104787.1-201"; exon_number 1; exon_id "ENSE00003759492.1"; level 3; protein_id "ENSP00000485388.1"; tag "basic"; transcript_support_level "5"; tag "appris_principal_1";
chr22	ENSEMBL	exon	11065974	11066015	.	-	.	gene_id "ENSG00000280363.1"; transcript_id "ENST00000623473.1"; gene_type "protein_coding"; gene_status "KNOWN"; gene_name "CU104787.1"; transcript_type "protein_coding"; transcript_status "KNOWN"; transcript_name "CU104787.1-201"; exon_number 2; exon_id "ENSE00003755067.1"; level 3; protein_id "ENSP00000485388.1"; tag "basic"; transcript_support_level "5"; tag "appris_principal_1";
chr22	ENSEMBL	CDS	11065974	11066015	.	-	0	gene_id "ENSG00000280363.1"; transcript_id "ENST00000623473.1"; gene_type "protein_coding"; gene_status "KNOWN"; gene_name "CU104787.1"; transcript_type "protein_coding"; transcript_status "KNOWN"; transcript_name "CU104787.1-201"; exon_number 2; exon_id "ENSE00003755067.1"; level 3; protein_id "ENSP00000485388.1"; tag "basic"; transcript_support_level "5"; tag "appris_principal_1";
```

**Comment by diekhans** [comment]
underlying problem is FASTA doesn't have UCSC naming, so sequence is not found.  Still, would be nice to have a friendlier report of the error.

**Comment by alexdobin** [comment]
Hi Mark,

if the GTF file contains transcript with names different from the FASTA, it's supposed to generate a warning in the Log.out file, not an error. I have seen this error before if the GTF entries were missing the "transcript_id" tags. I will try to reproduce this problem with your files.

Cheers
Alex

**Comment by diekhans** [comment]
Thanks Alex,

I believe the crash was cause because none of the transcript had
sequences in the genome fasts.  This resulted in accessing
uninitialized memory.

BTW, we are very positive around STAR at UCSC.  It's fast enough
to enable us to do a lot more processing with good results.

Cheers,
Mark

**Comment by alexdobin** [comment]
Hi Mark,

got it fixed finally in the 2.5.0a, should throw an error now.

Cheers
Alex
